### PR TITLE
Make `main.py` executable and create a symlink during installation

### PR DESCRIPTION
**Pull request by tmadlener**
BEGINRELEASENOTES
- Mark `main.py` as executable and create a symlink from `<prefix>/bin/k4GeneratorsConfig` to `<prefix>/python/main.py` during installation. This makes it possible to setup an environment where `k4GeneratorsConfig` can be used as a simple executable
- Remove some `PYTHONPATH` modifications from within the interpreter and use absolute imports. This makes it possible to simply add `<prefix>/pyhton` to the `PYTHONPATH` and everything will work.
  - Now we use relative imports internally mostly and make sure to expose `Generators.Generators` via settings in `__init__.py`

ENDRELEASENOTES

One thing I discovered that simply calling `<prefix>/bin/k4GeneratorsConfig` without argument, leads to:
```console
$ ./install/bin/k4GeneratorsConfig 
Generating and writing configuration files
Traceback (most recent call last):
  File "/home/tmadlener/work/key4hep/k4GeneratorsConfig/./install/bin/k4GeneratorsConfig", line 158, in <module>
    main()
  File "/home/tmadlener/work/key4hep/k4GeneratorsConfig/./install/bin/k4GeneratorsConfig", line 102, in main
    executeFiles(files, 0, rndmSeed, events)
  File "/home/tmadlener/work/key4hep/k4GeneratorsConfig/./install/bin/k4GeneratorsConfig", line 154, in executeFiles
    return rndmIncrement
           ^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'rndmIncrement' where it is not associated with a value
```

Ideally this should simply spit out a usage message, rather than error. Things work with `-h` or `--help`. Maybe there is an easy way to fix this, by e.g. making some of the arguments required, in which case `argparse` would do the right thing.

- [ ] name of `k4GeneratorsConfig` symlink? See also dicsussion in https://github.com/key4hep/key4hep-spack/issues/617
- [ ] Also update the setup.sh script?

### CODE REVIEW

**Comment by dirkzerwas** [comment]
Protection added against -f FILENAME absent on command line

**Comment by tmadlener** [comment]
Is not passing an input yaml file a "user error"? I.e. is the expectation that there should always be at least one yaml file? Because then we can just change `nargs="*"`, to `nargs="+"`.

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,9 +87,19 @@ configure_file(
...
+  DESTINATION ${CMAKE_INSTALL_PREFIX}
+  REGEX test_.*\\.py|main.py$ EXCLUDE  # Don't install python unittests
   PATTERN __pycache__ EXCLUDE  # Or pythons caches
 )
+# Install main as an executable
+install(PROGRAMS ${PROJECT_SOURCE_DIR}/python/main.py DESTINATION ${CMAKE_INSTALL_PREFIX}/python)
+
+install(CODE "execute_process(COMMAND bash -c \"set -e
+    link_target=${CMAKE_INSTALL_PREFIX}/python/main.py
+    link_name=${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/k4GeneratorsConfig
```
Comment (tmadlener) [inline CMakeLists.txt:99]:
```suggestion
    link_name=${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/k4generatorsConfig
```

I just realized that the `alias` in `setup.sh` is capitalized this way. I have no preference either way, it just has to be consistent in the end. @dirkzerwas any preferences from your side?

Comment (dirkzerwas) [reply CMakeLists.txt:99]: Your choice makes more sense, let's leave it as "k4GeneratorsConfig" , I will update the setups and tests to be consistent.

**Pull request merged**